### PR TITLE
[WIP] [RHCLOUD-17059] Add Authentications/ApplicationAuthentications to Records.event 

### PR DESCRIPTION
**Pull request by lpichler**
This PR is split to more smaller PRs:

- [x] 1 https://github.com/RedHatInsights/sources-api-go/pull/77
- [x] 2 https://github.com/RedHatInsights/sources-api-go/pull/78
- [x] 3 https://github.com/RedHatInsights/sources-api-go/pull/79
- [x] 4 https://github.com/RedHatInsights/sources-api-go/pull/80
- [x] 5  https://github.com/RedHatInsights/sources-api-go/pull/87
- [x] 6 https://github.com/RedHatInsights/sources-api-go/pull/98
- [x] 7 https://github.com/RedHatInsights/sources-api-go/pull/117

in ruby project:
https://github.com/RedHatInsights/sources-api/blob/19cf1aa2f6f2ba7f0ce8c17dc8a4662cfabc368d/app/models/concerns/event_concern.rb

Change adds https://github.com/RedHatInsights/sources-api/blob/19cf1aa2f6f2ba7f0ce8c17dc8a4662cfabc368d/app/models/concerns/event_concern.rb#L14-L25
and 
https://github.com/RedHatInsights/sources-api/blob/19cf1aa2f6f2ba7f0ce8c17dc8a4662cfabc368d/app/models/concerns/event_concern.rb#L39-L40

### Links
https://issues.redhat.com/browse/RHCLOUD-17059

### CODE REVIEW

File: util/xrh_header.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+	if err != nil {
+		return nil, fmt.Errorf("x-rh-identity header does not contain valid JSON")
+	}
+
+	return XRHIdentity, nil
+}
+
+func AccountNumberFrom(headers []kafka.Header) (string, error) {
+	XRHAccountNumberKey := "x-rh-sources-account-number"
+	XRHIdentityKey := "x-rh-identity"
```
Comment (MikelAlejoBR) [inline util/xrh_header.go:28]: I would put these on the package level as a couple of vars, so they don't get instantiated every time

**Review by MikelAlejoBR — APPROVED** [review]

**Comment by lpichler** [comment]
@MikelAlejoBR thanks for looking on it and it is possible that final code will be changed as partial PRs will be opened and reviewed

**Comment by MikelAlejoBR** [comment]
Aaaaah okay, I'll wait for the review request then :+1:

**Comment by lpichler** [comment]
all PRs merged

**Pull request closed without merging**